z3c/dependencychecker/package.py

```diff
@@ -0,0 +1,178 @@
+import os
+import pkg_resources
+import sys
+
+
+logger = logging.getLogger(__name__)
+
+
+class PackageMetadata(object):
+    """Information related to a python source distribution
+
+    It relies heavily on setuptools and pkg_resources APIs.
+    """
+
+    def __init__(self, path):
+        self.distribution_root = self._get_distribution_root(path)
+        self.setup_py_path = self._get_setup_py_path()
+        self.package_dir = self._get_package_dir()
+        self.egg_info_dir = self._get_egg_info_dir()
+        self.name = self._get_package_name()
+        self._working_set = self._generate_working_set_with_package()
+        self.top_level = self._get_sources_top_level()
+
+    @staticmethod
+    def _get_distribution_root(path):
+        if 'setup.py' not in os.listdir(path):
+            logger.error(
+                'setup.py was not found in %s. '
+                'Without it dependencychecker can not work.',
+                path,
+            )
+            sys.exit(1)
+        return path
+
+    def _get_setup_py_path(self):
+        return os.path.join(
+            self.distribution_root,
+            'setup.py',
+        )
+
+    def _get_package_dir(self):
+        """Check where the .egg-info is located"""
+        try_paths = (
+            self.distribution_root,
+            os.path.join(self.distribution_root, 'src', )
+        )
+        for path in try_paths:
+            folder_found = self._check_egg_in_folder(path)
+            if folder_found:
+                logger.debug(
+                    '.egg-info folder found at %s folder',
+                    path,
+                )
+                return path
+
+        logger.error('.egg-info folder could not be found')
+        sys.exit(1)
+
+    @staticmethod
+    def _check_egg_in_folder(path):
+        if not os.path.exists(path):
+            logger.error(
+                'Folder %s does not exist',
+                path,
+            )
+            sys.exit(1)
+
+        with change_dir(path):
+            results = glob.glob('*.egg-info')
+
+        return results
+
+    def _get_egg_info_dir(self):
+        results = self._check_egg_in_folder(self.package_dir)
+        return os.path.join(
+            self.package_dir,
+            results[0],
+        )
+
+    def _get_package_name(self):
+        path, filename = os.path.split(self.egg_info_dir)
+        name = filename[:-len('.egg-info')]
+        logger.debug(
+            'Package name is %s',
+            name,
+        )
+
+        return name
+
+    def _generate_working_set_with_package(self):
+        """Use pkg_resources API to enable the package being analyzed
+
+        This, enabling the package in pkg_resources, allows to extract
+        information from .egg-info folders, for example, its requirements.
+        """
+        working_set = pkg_resources.WorkingSet()
+        working_set.add_entry(self.package_dir)
+        return working_set
+
+    def _get_package_from_working_set(self):
```

Comment:
"package" is probably the original terminology? I'm asking because you return a "distribution". And the docstring mentions a "requirement". Three terms for the same thing.

I know from buildout that it can be quite confusing whether you have to deal with requirements or installed packages or.... => perhaps take a quick look if one or two of the names need changing.

Reply:
Yup, that was way too much, I removed distribution and changed it to package, I keep Requirement on the docstring as that's the object class one gets from pkg_resources.

And while at it I decided to rename two methods:
``_get_package_from_working_set`` -> ``_get_ourselves_from_working_set``
``_generate_working_set_with_package`` -> ``_generate_working_set_with_ourselves``

Looking at the method names it seemed to imply that an argument of them was a random package that you should give, but it is rather the very same package that we are dealing with, so I changed ``package`` to ``ourselves``

Reply:
Much clearer!